docs/modules/getting-started/pages/push.adoc

```diff
@@ -0,0 +1,112 @@
+
+[source,java]
+----
+PushService pushService = MobileCore.getInstance().getService(PushService.class);
+
+pushService.registerDevice(new Callback() {
+    @Override
+    public void onSuccess() {
+        // Yay
+    }
+
+    @Override
+    public void onError(Throwable error) {
+        // Ops!
+    }
+});
+----
+
+You also can add some optional parameters using `UnifiedPushConfig`
+
+[source,java]
+----
+UnifiedPushConfig unifiedPushConfig = new UnifiedPushConfig();
+unifiedPushConfig.setAlias("AeroGear");
+unifiedPushConfig.setCategories(Arrays.asList("Android", "OpenShift"));
+
+pushService.registerDevice(unifiedPushConfig, yourCallback);
+----
+
+If you want to add option for user to stop receiving notifications.
+
+[source,java]
+----
+PushService pushService = MobileCore.getInstance().getService(PushService.class);
+pushService.unregisterDevice(new Callback() {
+    @Override
+    public void onSuccess() {
+        // yay
+    }
+
+    @Override
+    public void onError(Throwable error) {
+        // Ops!
+    }
+});
+----
+
+== Handling messages
+
+`MessageHandler` is responsable for handling the messages received from the FCM. It will give you total flexibility to decide what to do with the message recieved.
+
+## Receiving Background and Foreground messages
+
+Currently, push messages are received by an instance of  `AeroGearFirebaseMessagingService`. The messages are processed and passed to `PushService.notifyHandlers`.
+
+Classes which implement `MessageHandler` can ask that their methods be called on either the UI thread using `PushService.registerMainThreadHandler` or on a background thread using `PushService.registerBackgroundThreadHandler`. UI threads are most useful for Activities and Fragments; background threads are most useful for everything else.
+
+In an Activity (or in a Fragment) you must remove the handler when the Activity goes into the background and must reenable it when it comes into the foreground.
+
+[source,java]
+----
+@Override
+public void onStart() {
+    super.onStart();
+    PushService.registerMainThreadHandler(this);
+    PushService.unregisterBackgroundThreadHandler(myBackgrondHandlerInstance);
+}
+
+@Override
+public void onStop() {
+    super.onStop();
+    PushService.unregisterMainThreadHandler(this);
+    PushService.registerBackgroundThreadHandler(myBackgrondHandlerInstance);
+}
+----
+
+== Default handler
+
+Unfortunately when you close or kill the app the Android system can destroy the `PushService` instance and we will lose the handler you have added there. To solve this we provide a way to register a default handler, it will be used when app is not running (actually running in background) and there are no others handlers registered.
+
+[source,xml]
+----
+<meta-data
+    android:name="DEFAULT_MESSAGE_HANDLER_KEY"
+    android:value="package.ClassName" />
+----
```

Comment:
There are two details missing.  1) This stanza has to be in the application block and 2) this has to be in AndroidManifest.xml